demos/logging-stack/logging_stack.h

```diff
@@ -41,7 +41,11 @@
 
 /* Common macro for all logging interface macros. */
 /* TODO - Replace printf with an implementation function. */
```

Comment:
Who is this `TODO` targeted to?  A developer _of_ this library, or a developer _who uses_ this library?

Reply:
For us, since printf may not be thread-safe on all POSIX compliant MCUs. Archit has a PoC for using a logging thread for log message inputs

Reply:
Then I'd make sure that it is clear to anyone using this library that it's not their responsibility to implement that `TODO`.  I think we need some more explanation here.

Reply:
The TODO is just temporary until the PR for the logging servicing thread is raised. If there is concern about it being misinterpreted, we can remove it completely for now.